Controller/RoleController.php

```diff
@@ -113,14 +113,35 @@ public function updateRoleAction(Request $request, $roleId)
         }
 
         try {
-            $role = $this->roleService->loadRole($roleId);
+            $roleDraft = $this->roleService->loadRoleDraft($roleId);
         } catch (NotFoundException $e) {
-            $this->notifyError('role.error.role_not_found', [], 'role');
+            // The draft doesn't exist, let's create one
+            try {
+                $role = $this->roleService->loadRole($roleId);
+            } catch (UnauthorizedException $e) {
+                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+            } catch (NotFoundException $e) {
+                $this->notifyError('role.error.role_not_found', [], 'role');
+
+                return $this->redirectToRoute('admin_roleList');
+            }
 
-            return $this->redirectToRoute('admin_roleList');
+            try {
+                $roleDraft = $this->roleService->createRoleDraftByRole($role);
+            } catch (UnauthorizedException $e) {
+                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+            } catch (InvalidArgumentException $e) {
+                $this->notifyError('role.error.has_draft', [], 'role');
+
+                return $this->redirectToRoute('admin_roleList');
+            } catch (LimitationValidationException $e) {
```

Comment:
This catch block is also useless since you don't create a role draft with policies.